### PR TITLE
Docs: clarify isDefault usage for registerBlockStyles()

### PR DESCRIPTION
**Pull request by ajitbohra**
~Removes `isDefault` info for `registerBlockStyle()` from handbook.~

Clarify isDefault usage for registerBlockStyles()

Reference https://github.com/WordPress/gutenberg/issues/11451

### CODE REVIEW

**Comment by tofumatt** [comment]
Hi @ajitbohra, thanks for the PR. In the future, feel free to assign more folks than just @chrisvanpatten and I to review documentation PRs. It's probably best to assign the review to gutenberg-core, so that a wider group of people can see the review request 😄

**Review by tofumatt — CHANGES_REQUESTED** [review]
I can't tell from the issue if there is a bug here or not; in the issue it seems like @dgwyer is saying this _should_ work. So I don't think removing the docs is the right approach.

**Comment by dgwyer** [comment]
I think it makes sense to have the `isDefault` property in `registerBlockStyle()` override the default style variation that may have been set in the blocks `registerBlockType()` function.

The issue is that `isDefault` appears to be ignored when used in `registerBlockStyle()`.

**Comment by tofumatt** [comment]
Yeah, that makes sense to me too.

**Comment by youknowriad** [comment]
I think people are confusing "isDefault" as the default style variation applied when you insert the block (concept we don't have at the moment). That's not what the current "isDefault" refer to. The current "isDefault" is used to give a name to the style variation that doesn't add any custom className. (The no-style style variation).

**Comment by ajitbohra** [comment]
@tofumatt noted agree with you on that

@youknowriad yes currently it's confusing if I am not wrong its kind of fallback style in case of missing style variation?

Instead of removing we need copy update for the `isDefault`

**Comment by dgwyer** [comment]
@youknowriad I think I'm more confused now than when I started. Here's what I found after more testing.

It appears that when `isDefault` is used inside `registerBlockStyle()` it's recognized ONLY if no styles in `registerBlockType()` have been set as the default. Otherwise `isDefault` is always ignored inside of `registerBlockStyle()`.

Also, even if a style variation is selected in the block UI on insertion no CSS class is actually added unless a style variation option is actually clicked on.

This feels counter intuitive to me. I'd expect the style variation CSS class to be applied whenever one is selected, either on block insertion or manual selection. This isn't currently the case.

And allowing `registerBlockStyle()` to override the default selected style variation would make a lot of sense.

**Comment by youknowriad** [comment]
@dgwyer yes, at the moment `isDefault` is just used to give a name the "no class" style variation. It's not meant to solve the style variation selected when you inserter the block use-case, this "concept" doesn't exist yet.

**Comment by gziolo** [comment]
@ajitbohra can you clarify the intent of `isDefault` instead of removing this line based on this comment from @youknowriad:

> The current "isDefault" is used to give a name to the style variation that doesn't add any custom className. (The no-style style variation).

**Comment by ajitbohra** [comment]
@gziolo PR updated to clarify the the usage of `isDefault`

**Comment by gziolo** [comment]
@youknowriad - is it correct now?

File: docs/designers-developers/developers/filters/block-filters.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ wp.blocks.registerBlockStyle( 'core/quote', {
 
 The example above registers a block style variation named `fancy-quote` to the `core/quote` block. When the user selects this block style variation from the styles selector, an `is-style-fancy-quote` className will be added to the block's wrapper.
 
-By adding `isDefault: true`, you can make registered style variation to be active by default when a block is inserted.
+By adding `isDefault: true` you can make the registered style variation to be used as a fallback if no style in `registerBlockType()` has been set as the default.
```
Comment (youknowriad) [inline docs/designers-developers/developers/filters/block-filters.md]: Maybe something like?

> By adding `isDefault: true` you can mark the registered style variation as the one that is active without any custom className. 

(Or something in that vein, it's hard to explain :P)

Comment (gziolo) [inline docs/designers-developers/developers/filters/block-filters.md]:
```suggestion
By adding `isDefault: true` you can mark the registered style variation as the one that is recognized as active when no custom class name is provided. It also means that there will be no custom class name added to the HTML output for the style that is marked as default.
```

Comment (gziolo) [reply docs/designers-developers/developers/filters/block-filters.md]: How about the one I tried to rephrase? 😅

Comment (youknowriad) [reply docs/designers-developers/developers/filters/block-filters.md]: I like it.

Comment (ajitbohra) [reply docs/designers-developers/developers/filters/block-filters.md]: looks good and clear to understand :+1:

**Review by gziolo — APPROVED** [review]
Haha, so @ajitbohra couldn't approve. Let me do it on his behalf as I authored the last version. Thanks for good teamwork on this one 💯

**Pull request merged**